### PR TITLE
Add template lock attribute to column and group

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR adds a templateLock attribute to columns and group. When this attribute is set the locking of the blocks is set to the attribute value instead of relying on the inherited value.

This allows the locking of these blocks to be controlled by templates, patterns, etc.

supersedes: https://github.com/WordPress/gutenberg/pull/25869

## How has this been tested?
I verified now locking is not inherited by using sample cpt templates, and sample blocks with inner block templates.

I verified I could control the locking of the group block by using the templateLock attribute.
By pasting the following code on the code editor I verified that the first group block is not locked, while the second group block is locked.

```
<!-- wp:group -->
<div class="wp-block-group"><div class="wp-block-group__inner-container"><!-- wp:quote -->
<blockquote class="wp-block-quote"><p></p></blockquote>
<!-- /wp:quote --></div></div>
<!-- /wp:group -->

<!-- wp:group {"templateLock":"all"} -->
<div class="wp-block-group"><div class="wp-block-group__inner-container"><!-- wp:quote -->
<blockquote class="wp-block-quote"><p></p></blockquote>
<!-- /wp:quote --></div></div>
<!-- /wp:group -->
```

I verified I could control the locking of a column by pasting the following code on the code editor I verified column 1 is locked while the second column is not.

```
<!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:column {"templateLock":"all"} -->
<div class="wp-block-column"><!-- wp:paragraph -->
<p>1</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class="wp-block-column"><!-- wp:paragraph -->
<p>2</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->
```

### CODE REVIEW

File: packages/e2e-tests/plugins/cpt-locking.php [hunk 1]
```diff
@@ -52,6 +52,82 @@ function gutenberg_test_cpt_locking() {
...
+					'core/group',
+					array(
+						'templateLock' => false,
+					),
+					array(
+						array( 'core/quote' ),
+						array(
+							'core/paragraph',
+							array(
+								'placeholder' => 'Add a description343',
```
Comment (mcsf) [inline packages/e2e-tests/plugins/cpt-locking.php:72]: Intentional or typo?

```suggestion
								'placeholder' => 'Add a description',
```

File: packages/e2e-tests/plugins/cpt-locking.php [hunk 2]
```diff
@@ -52,6 +52,82 @@ function gutenberg_test_cpt_locking() {
 			'template_lock' => false,
 		)
 	);
+	register_post_type(
+		'ul-post-ul-group',
```
Comment (mcsf) [inline packages/e2e-tests/plugins/cpt-locking.php:56]: If I understood the nomenclature, this should be:

```suggestion
		'l-post-ul-group',
```

File: packages/e2e-tests/plugins/cpt-locking.php (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/e2e-tests/plugins/cpt-locking.php]: It was a string I changed to make sure the code executing was the expected one and I missed o revert the change.

File: packages/e2e-tests/plugins/cpt-locking.php (hunk 2 above)
Comment (jorgefilipecosta) [reply packages/e2e-tests/plugins/cpt-locking.php]: Correct the string was updated.

Comment (jorgefilipecosta) [reply packages/e2e-tests/plugins/cpt-locking.php]: We are using this nomenclature because of the char limits post type id has.

**Comment by jorgefilipecosta** [comment]
Hi @mcsf, your feedback was applied.

**Review by mcsf — APPROVED** [review]
LGTM, though second opinions are welcome. :)

File: packages/e2e-tests/plugins/cpt-locking.php (hunk 1 above)
Comment (mcsf) [reply packages/e2e-tests/plugins/cpt-locking.php]: Hah, it's the same reason for my adding `die( 42 );` in the other PR today. :D

File: packages/block-library/src/column/block.json [hunk 3]
```diff
@@ -11,6 +11,9 @@
 		},
 		"width": {
 			"type": "string"
+		},
+		"templateLock": {
+			"type": "string"
```
Comment (ZebulanStanphill) [inline packages/block-library/src/column/block.json:16]: Shouldn't this be a boolean?

Comment (jorgefilipecosta) [reply packages/block-library/src/column/block.json:16]: Hi @ZebulanStanphill, the template lock attribute can contain values like "all", and "insert" which are strings.

**Pull request merged**